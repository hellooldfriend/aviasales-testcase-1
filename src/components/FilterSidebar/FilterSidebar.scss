@import '../../variables.scss';

.filter_sidebar {
    width: 232px; // TODO: remove
    display: inline-block;
    background: $background_secondary;
    border-radius: $border_radius_basic;
    box-sizing: border-box;
    padding-top: 18px;
    padding-bottom: 10px;

    &-title {
        box-sizing: border-box;
        padding-left: $p2;
        padding-right: $p2;
        letter-spacing: 0.65px;
        text-transform: uppercase;
        margin-bottom: 6px;
    }
}