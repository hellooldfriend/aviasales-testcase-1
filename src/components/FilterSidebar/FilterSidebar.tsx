import React from 'react'
import './FilterSidebar.scss'

import Checkbox from '../Checkbox'
import { IFilterSidebar } from '../../types'


const OPTIONS = [
    {
        code: 0,
        title: 'Без пересадок',
    },
    {
        code: 1,
        title: '1 пересадка',
    },
    {
        code: 2,
        title: '2 пересадки',
    },
    {
        code: 3,
        title: '3 пересадки',
    }
]

const FilterSidebar = (props: IFilterSidebar) => {
    const activeStops = props.activeStops || []
    const stops = props.stops || []

    const allActive = activeStops.length === OPTIONS.length

    return (
        <div className="filter_sidebar"
            data-testid="filter_sidebar"
        >
            <div className="filter_sidebar-title">Количество пересадок</div>

            <div className="filter_sidebar-items">
                <Checkbox
                    value={allActive}
                    title={'Все'}
                    onChange={() => {
                        if(allActive) {
                            props.onAllStopsClick([])
                        } else {
                            const values = OPTIONS.map(option => option.code)
                            props.onAllStopsClick(values)
                        }
                    }}
                />

                {OPTIONS.map(option => {
                    return (
                        <Checkbox
                            key={option.code}
                            value={activeStops.includes(option.code)}
                            disabled={!stops.includes(option.code)}
                            title={option.title}
                            onChange={() => props.onChange(option.code)}
                        />
                    )
                })}
            </div>

        </div>
    )
}


export default FilterSidebar
