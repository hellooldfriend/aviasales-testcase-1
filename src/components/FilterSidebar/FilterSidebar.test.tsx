import React from 'react'
import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

import FilterSidebar from './FilterSidebar'


afterEach(() => {
    cleanup()
})


const props = {
    stops: [],
    activeStops: [],
}

describe('FilterSidebar', () => {
    test('Should render', () => {
        render(
            <FilterSidebar
                stops={props.stops}
                activeStops={props.activeStops}
            />
        )
        const element = screen.getByTestId('filter_sidebar')
        expect(element).toBeInTheDocument()
    })

    test('Matches snapshot', () => {
        const element = renderer.create(
            <FilterSidebar
                stops={props.stops}
                activeStops={props.activeStops}
            />
        ).toJSON()
        expect(element).toMatchSnapshot()
    })
})