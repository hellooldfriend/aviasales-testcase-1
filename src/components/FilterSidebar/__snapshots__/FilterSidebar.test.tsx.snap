// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSidebar Matches snapshot 1`] = `
<div
  className="filter_sidebar"
  data-testid="filter_sidebar"
>
  <div
    className="filter_sidebar-title"
  >
    Количество пересадок
  </div>
  <div
    className="filter_sidebar-items"
  >
    <label
      className="checkbox"
      data-testid="checkbox"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      Все
    </label>
    <label
      className="checkbox _disabled"
      data-testid="checkbox"
    >
      <input
        checked={false}
        disabled={true}
        onChange={[Function]}
        type="checkbox"
      />
      Без пересадок
    </label>
    <label
      className="checkbox _disabled"
      data-testid="checkbox"
    >
      <input
        checked={false}
        disabled={true}
        onChange={[Function]}
        type="checkbox"
      />
      1 пересадка
    </label>
    <label
      className="checkbox _disabled"
      data-testid="checkbox"
    >
      <input
        checked={false}
        disabled={true}
        onChange={[Function]}
        type="checkbox"
      />
      2 пересадки
    </label>
    <label
      className="checkbox _disabled"
      data-testid="checkbox"
    >
      <input
        checked={false}
        disabled={true}
        onChange={[Function]}
        type="checkbox"
      />
      3 пересадки
    </label>
  </div>
</div>
`;
